kernel/futex.c

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2019 Intel corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <kernel_structs.h>
+#include <spinlock.h>
+#include <kswap.h>
+#include <wait_q.h>
+#include <syscall_handler.h>
+#include <init.h>
+#include <ksched.h>
+
+static struct k_spinlock lock;
```

Comment:
That seems a reasonable change.

Reply:
There should be enough memory if enabled user space. But I don't think there is lock breaking for single core system.  And I will add separate spinlock for each futex object because we support SMP.